cmd/contour/serve.go

```diff
@@ -278,14 +278,20 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	}
 	g.Add(lbsw.Start)
 
-	// step 12. register an informer to watch envoy's service.
-	ssw := &k8s.ServiceStatusLoadBalancerWatcher{
-		ServiceName: ctx.envoyServiceName,
-		LBStatus:    lbsw.lbStatus,
+	// step 12. register an informer to watch envoy's service if we haven't been given static details.
+	if ctx.ingressStatusAddress == "" {
```

Comment:
Do we care if the address passed in is valid? Like if I entered `steve.s.%%%` for the address. Maybe a followup issue, but just thinking out loud.

Reply:
We could check that it's either a DNS name or IP address, I suppose.

Reply:
Actually, I've thought over this more, and I don't think it's worth doing much validation here. We already check if it's an IP or a hostname, I think that's enough.